internal/pkg/workspace/workspace.go

```diff
@@ -160,6 +162,66 @@ func (ws *Workspace) ListWorkloads() ([]string, error) {
 	})
 }
 
+// Pipeline holds identifying information about a pipeline.
+type Pipeline struct {
+	Name string
+	Path string
+}
+
+// ListPipelines returns all pipelines in the workspace.
+func (ws *Workspace) ListPipelines() ([]Pipeline, error) {
+	var pipelines []Pipeline
+	// Look for legacy pipeline.
+	legacyPath, err := ws.PipelineManifestLegacyPath()
+	if err != nil {
+		return nil, err
+	}
+	manifest, err := ws.ReadPipelineManifest(legacyPath)
+	if err != nil {
+		log.Infof("Unable to read pipeline manifest at '%s'", legacyPath)
+	}
+	pipelines = append(pipelines, Pipeline{
+		Name: manifest.Name,
+		Path: legacyPath,
+	})
```

Comment:
Right now if `ReadPipelineManifest` fails because the file doesn't exist, then we still append a `Pipeline` with `manifest.Name = ""`.

How about the following:
```go
var manifests []PipelineManifest

legacyMft, err := ws.legacyPipelineManifest()
if err != nil && !errors.Is(err, ErrNoPipelineInWorkspace) {
    return err
}
if err == nil {
   manifests = append(manifests, legacyMft)
}
```